druid/src/window.rs

```diff
@@ -154,6 +154,18 @@ impl<T: Data> Window<T> {
                 false,
             );
         }
+
+        if let Some(focus_req) = widget_state.request_focus.take() {
+            let old = self.focus;
+            let new = self.widget_for_focus_request(focus_req);
+            // Only send RouteFocusChanged in case there's actual change
+            if old != new {
+                let event = LifeCycle::Internal(InternalLifeCycle::RouteFocusChanged { old, new });
+                self.lifecycle(queue, &event, data, env, false);
+                self.focus = new;
+            }
+        }
+
```

Comment:
Ugh, this is complicated. 😣

We will need to have handling in here to send the `EnabledChanged` event, and then we will need to update the focus chain; I think what makes sense then is to check if the currently focused widget still exists, and to only send a `FocusChanged` if it doesn't.

Reply:
Im not sure i understand what you mean. Are you talking about the situation when a widget calls `request_focus()` or the focus change issued in `update_enabled()`?

Reply:
I'm also not sure I understand what I meant.

I think what I was concerned about, here, is the case where the widget that was currently focused becomes disabled, and thus is no longer in the focus chain. In this case we still need to send the `RouteFocusChanged` event, because we need to update every ancestor of that widget so that they update `has_focus`. Thinking about this more, we also need to send `FocusChanged` to the `WidgetPod` in question (to update its `is_focused` flag) but we *don't* (I think) need to send this message to the *actual widget*, which should have already been disabled, which implies a loss of focus. 🤔

Reply:
My current Aproach is to set `focus_change_request` to `Resign` if a widget which contains the focused element disables it self. Doing this from the root would be difficult since we don't know which Widgets are allowed to be focused. Should I keep it that way?

Reply:
I think your approach should work.

And it's a good point that we don't know what widgets can be focused; we only know what widgets participate in the focus chain, which is not *exactly* the same thing.